Managers/UInputMan.cpp

```diff
@@ -134,6 +133,39 @@ namespace RTE {
 				return true;
 			}
 		}
+#elif defined __unix__
+		int numDetectedJoysticks = 0;
+		std::string devName = "/dev/input/jsX";
+		std::string devNameFallback = "/dev/jsX";
+		for (numDetectedJoysticks = 0; numDetectedJoysticks < MAX_JOYSTICKS; ++numDetectedJoysticks) {
+			devName.at(devName.length() - 1) = '0' + numDetectedJoysticks;
+			devNameFallback.at(devNameFallback.length() - 1) = '0' + numDetectedJoysticks;
+
+			// Note - Allegro only checks until the first missing joystick, so there's no point in checking any more than that.
+			if (!std::filesystem::exists(devName) && !std::filesystem::exists(devNameFallback)) {
+				break;
+			}
+		}
+
+		if (numDetectedJoysticks != num_joysticks) {
+			if (numDetectedJoysticks > 0) {
+				for (int i = 0; i < numDetectedJoysticks; ++i) {
+					devName.at(devName.length() - 1) = '0' + i;
+					devNameFallback.at(devNameFallback.length() - 1) = '0' + i;
+
+					int joystickDescriptor = open(devName.c_str(), O_RDONLY | O_NONBLOCK);
```

Comment:
Can we not use std::open here instead of c-style open?

Reply:
std::open doesn't exist. `open` is a linux io thing, I'll also need to recheck that the relevant headers are included explicitly.

Reply:
Ah yeah, I was thinking of fopen lol. Thanks :D